src/main/java/com/pikachu/purple/application/user/vo/NicknameCreator.java

```diff
@@ -0,0 +1,16 @@
+package com.pikachu.purple.application.user.vo;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@Getter
+@RequiredArgsConstructor
+public class NicknameCreator {
+
+    private final String nickName;
+
+    public static NicknameCreator create(int countTotalUsers){
+        String createNickname = String.format("퍼픽%03d", countTotalUsers+1);
+        return new NicknameCreator(createNickname);
+    }
+}
```

Comment:
p3) 추가적으로 아까 언급한 사항이긴 한데, NicknameCreator에서 create의 결과가 NicknameCreator를 그대로 반환하는 건 어색한 것 같습니다 🙂
```suggestion
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public class NicknameCreator {

    private static final String NICKNAME_FORMAT = "퍼픽%03d";
    private final String nickname;

    public static NicknameCreator create(int userTotalCount){
        String createdNickname = String.format(NICKNAME_FORMAT, countTotalUsers+1);
        return new NicknameCreator(createdNickname);
    }

}
```

Reply:
해결 완료!